src/Compilers/CSharp/Portable/Symbols/EventSymbol.cs

```diff
@@ -258,14 +261,15 @@ internal EventSymbol AsMember(NamedTypeSymbol newOwner)
         {
             Debug.Assert(this.IsDefinition);
             Debug.Assert(ReferenceEquals(newOwner.OriginalDefinition, this.ContainingSymbol.OriginalDefinition));
-            return newOwner.IsDefinition ? this : new SubstitutedEventSymbol(newOwner as SubstitutedNamedTypeSymbol, this);
+            Debug.Assert(newOwner.IsDefinition || newOwner is SubstitutedNamedTypeSymbol);
+            return newOwner.IsDefinition ? this : new SubstitutedEventSymbol((newOwner as SubstitutedNamedTypeSymbol)!, this);
```

Comment:
>(newOwner as SubstitutedNamedTypeSymbol)! [](start = 77, length = 41)

Should be a cast then?

Reply:
I couldn't confirm statically that this conversion is always non-null, and didn't want to be the cause of a new exception.

Reply:
➡️ okay